salt/config.py

```diff
@@ -1411,7 +1412,10 @@ def _read_conf_file(path):
             conf_opts = {}
         # allow using numeric ids: convert int to string
         if 'id' in conf_opts:
-            conf_opts['id'] = str(conf_opts['id'])
+            if not isinstance(conf_opts['id'], six.string_types):
+                conf_opts['id'] = str(conf_opts['id'])
```

Comment:
We've not verified that it's string_types, we've verified that it's not string_types.  If it is string_types, it is promoted to unicode, as @cachedout indicates.